src/nft/components/profile/list/RoyaltyTooltip.tsx

```diff
@@ -1,27 +1,87 @@
 import { Trans } from '@lingui/macro'
-import { ListingMarket } from 'nft/types'
-// eslint-disable-next-line no-restricted-imports
+import Column from 'components/Column'
+import Row from 'components/Row'
+import { ListingMarket, WalletAsset } from 'nft/types'
+import { formatEth } from 'nft/utils'
 import styled from 'styled-components/macro'
 import { ThemedText } from 'theme'
 
-const FeeWrap = styled.div`
+const FeeWrap = styled(Row)`
   margin-bottom: 4px;
-  color: ${({ theme }) => theme.textPrimary};
+  justify-content: space-between;
 `
 
-const RoyaltyContainer = styled(ThemedText.BodySmall)`
-  margin-bottom: 8px;
+const RoyaltyContainer = styled(Column)`
+  gap: 12px;
+  padding 4px 0px;
 `
 
-export const RoyaltyTooltip = ({ selectedMarket }: { selectedMarket: ListingMarket }) => {
+const MarketIcon = styled.img`
+  width: 16px;
+  height: 16px;
+  border-radius: 2px;
+  object-fit: cover;
+  outline: 1px solid ${({ theme }) => theme.backgroundInteractive};
+  margin-right: 8px;
+`
+
+const CollectionIcon = styled(MarketIcon)`
+  border-radius: 50%;
+`
+
+const FeePercent = styled(ThemedText.Caption)`
+  line-height: 16px;
+  color: ${({ theme }) => theme.textSecondary};
+  white-space: nowrap;
+`
+
+const MaxFeeContainer = styled(Row)`
+  justify-content: space-between;
+  padding-top: 12px;
+  border-top: 1px solid ${({ theme }) => theme.backgroundOutline};
+`
+
+export const RoyaltyTooltip = ({
+  selectedMarkets,
+  asset,
+  fees,
+}: {
+  selectedMarkets: ListingMarket[]
+  asset: WalletAsset
+  fees?: number
+}) => {
+  const maxRoyalty = Math.max(...selectedMarkets.map((market) => market.royalty ?? 0))
   return (
-    <RoyaltyContainer key={selectedMarket.name}>
-      <FeeWrap>
-        {selectedMarket.name}: {selectedMarket.fee}%
-      </FeeWrap>
+    <RoyaltyContainer>
+      {selectedMarkets.map((market) => (
+        <FeeWrap key={asset.collection?.address ?? '' + asset.tokenId + market.name + 'fee'}>
+          <Row>
+            <MarketIcon src={market.icon} />
+            <ThemedText.Caption lineHeight="16px" marginRight="12px">
+              {market.name}&nbsp;
+              <Trans>fee</Trans>
+            </ThemedText.Caption>
+          </Row>
+          <FeePercent>{market.fee}%</FeePercent>
+        </FeeWrap>
+      ))}
       <FeeWrap>
-        <Trans>Creator royalties</Trans>: {selectedMarket.royalty}%
+        <Row>
+          <CollectionIcon src={asset.collection?.imageUrl} />
+          <ThemedText.Caption lineHeight="16px" marginRight="12px">
+            <Trans>Max creator royalties</Trans>
+          </ThemedText.Caption>
+        </Row>
+        <FeePercent>{maxRoyalty}%</FeePercent>
       </FeeWrap>
+      <MaxFeeContainer>
+        <ThemedText.Caption lineHeight="16px">
+          <Trans>Max fees</Trans>
+        </ThemedText.Caption>
+        <ThemedText.Caption lineHeight="16px" color={fees ? 'textPrimary' : 'textSecondary'}>
+          {fees ? formatEth(fees) : '-'} ETH
```

Comment:
the number formatter util will work for this I think

Reply:
the formatter unfortunately doesn't have the rounding to K or M for large numbers
ethNumberStandardFormatter
<img width="220" alt="Screen Shot 2023-02-15 at 14 03 51 " src="https://user-images.githubusercontent.com/11512321/219182666-b4919707-91ba-4971-a67e-6b3e2b7190c4.png">
formatEth
<img width="240" alt="Screen Shot 2023-02-15 at 14 04 05 " src="https://user-images.githubusercontent.com/11512321/219182678-aebd5d3a-3c32-4c8a-a3d7-8dbb76eb2187.png">